File: src/automatica.core/Automatica.WebNew/src/app/base/model/areas/area-type.ts
import { Model, BaseModel, JsonFieldInfo, JsonProperty } from "../base-model";

@Model()
export class AreaType extends BaseModel {

    @JsonProperty()
    ObjId: string;

    @JsonProperty()
    Name: string;

    @JsonProperty()
    Description: string;

    public typeInfo(): string {
        return "AreaType";
    }

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return void 0;
    }
}

File: src/automatica.core/Automatica.Core.EF/Models/Areas/AreaType.cs
using System;
using Automatica.Core.Model;

namespace Automatica.Core.EF.Models.Areas
{
    public class AreaType : TypedObject
    {
        public Guid ObjId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
    }
}
